Source/Api/Controllers/AuthController.cs

```diff
@@ -95,33 +98,45 @@ public class AuthController : ExceptionlessApiController {
             }
 
             User user;
-            try {
-                user = await _userRepository.GetByEmailAddressAsync(model.Email);
-            } catch (Exception ex) {
-                _logger.Error().Exception(ex).Critical().Message("Login failed for \"{0}\": {1}", model.Email, ex.Message).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
-                return Unauthorized();
-            }
+	        if (Settings.Current.EnableActiveDirectoryAuth && !IsAdLoginValid(model.Email, model.Password)) {
```

Comment:
I'd move this check to after the user not null check, and then just return the token logic if it's valid otherwise return unauthorized.

Also on a side note, could you install the https://visualstudiogallery.msdn.microsoft.com/410e9b9f-65f3-4495-b68e-15567e543c58 extension, I noticed spacing and bracing styles are a different and we are trying to stay with k&r style and this plugin will help automatically apply our styles :)